### DISCUSSION TITLE
Allow modification of export formats for element exports

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/cms#10104] [Ideas] [answered]
Currently, JSON, XML and CSV are hard-coded as the available formats. It'd be great to be able to override these, or extend them.

https://github.com/craftcms/cms/blob/7b7f033a213bd1e9dbe1e9475863a7ec8f609e54/src/web/assets/cp/src/js/BaseElementIndex.js#L1953-L1959

One such use-case would be to export as TSV, YAML, or XLSX (same as CSV, but see https://github.com/verbb/formie/issues/638)

**Comment by mnlmaier** [comment]
@brandonkelly — do you know if this is something which might be implemented? we need Excel export functionality for a client's Formie submissions, and as Formie is relying on native element exports, we are wondering how to approach this.

**Comment by christophpaster** [comment]
@brandonkelly - this would be extremely helpful, basically for the same reason as linked above: XLSX-Export for Formie https://github.com/verbb/formie/issues/638

**Comment by brandonkelly** [comment]
Seems like it would make more sense for Craft to just add XSLX as a supported export format for all element types?

**Comment by brandonkelly** [comment, marked as answer]
Adde XLSX + YAML options for Craft 5.9 (#18160).

**Reply by engram-design** [reply]
Cheers @brandonkelly ideally it'd be nice to be able to customise these so we can extend them, but this will hopefully do. I know exporting in general will be getting an overhaul at some point too.

**Reply by brandonkelly** [reply]
Yeah, lots of changes to exporting in the works for v6, so doesn’t make sense to make it extensible in v5.

**Reply by mnlmaier** [reply]
@brandonkelly — awesome, thanks so much! I agree with @engram-design that, at some point, custom exporters would be a great feature, but for now, we can work with the new options.

**Reply by brandonkelly** [reply]
Craft 5.9.0 is out now with XLSX and YAML support 🎉
